src/lib/io/sol-ipm.c

```diff
@@ -0,0 +1,323 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <errno.h>
+#include <inttypes.h>
+#include <stdlib.h>
+
+#ifndef SOL_LOG_DOMAIN
+#define SOL_LOG_DOMAIN &_sol_ipm_log_domain
+#endif
+
+#include "sol-log-internal.h"
+
+#include "ipm.h"
+#include "ipm/ipm_quark_se.h"
+
+#include "sol-log.h"
+#include "sol-macros.h"
+#include "sol-mainloop-zephyr.h"
+#include "sol-types.h"
+#include "sol-util.h"
+#include "sol-vector.h"
+
+SOL_LOG_INTERNAL_DECLARE_STATIC(_sol_ipm_log_domain, "ipm");
+
+#define SOL_IPM_ID_FREE_REMOTE 0
+
+/* Declare channels we are going to use.
+ * Channel 0 is Quark -> ARC
+ * Channel 1 is ARC -> Quark
+ */
+#ifdef CONFIG_X86
+QUARK_SE_IPM_DEFINE(message_ipm_receiver, 1, QUARK_SE_IPM_INBOUND);
+QUARK_SE_IPM_DEFINE(message_ipm_sender, 0, QUARK_SE_IPM_OUTBOUND);
```

Comment:
On 06-05-2016 16:58, Iván Briano wrote:
> In src/lib/io/sol-ipm.c
> <https://github.com/solettaproject/soletta/pull/1981#discussion_r62382942>:
>
>> +#include "sol-mainloop-zephyr.h"
>> +#include "sol-types.h"
>> +#include "sol-util.h"
>> +#include "sol-vector.h"
>> +
>> +SOL_LOG_INTERNAL_DECLARE_STATIC(_sol_ipm_log_domain, "ipm");
>> +
>> +#define SOL_IPM_ID_FREE_REMOTE 0
>> +
>> +/* Declare channels we are going to use.
>> + * Channel 0 is Quark -> ARC
>> + * Channel 1 is ARC -> Quark
>> + */
>> +#ifdef CONFIG_X86
>> +QUARK_SE_IPM_DEFINE(message_ipm_receiver, 1, QUARK_SE_IPM_INBOUND);
>> +QUARK_SE_IPM_DEFINE(message_ipm_sender, 0, QUARK_SE_IPM_OUTBOUND);
>
> Make the channels configurable, even if it's through Kconfig. We can't
> assume that something else in the OS will not have taken these already.

Fair.

>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/solettaproject/soletta/pull/1981/files/4e9ba992a37f1d26c07d86c0fd12a2e9b93605be..214c2bc23bc15f893f48c03a0863ec03a614fe5f#r62382942>
>